staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -967,7 +968,8 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 	requestKey := requestIndices{claimIndex: r.claimIndex, requestIndex: r.requestIndex, subRequestIndex: r.subRequestIndex}
 	requestData := alloc.requestData[requestKey]
 	request := requestData.request
-	if !request.adminAccess() && (alloc.allocatedDevices.Has(device.id) || alloc.allocatingDevices[device.id]) {
+	if !request.adminAccess() && (alloc.allocatedDevices.Has(device.id) || alloc.allocatingDeviceForAnyClaim(device.id)) ||
+		alloc.allocatingDeviceForClaim(device.id, r.claimIndex) {
```

Comment:
Third check seems to be redundant:
```suggestion
	if !request.adminAccess() && (alloc.allocatedDevices.Has(device.id) || alloc.allocatingDeviceForAnyClaim(device.id)) {
```